docs/migration-guide.md

```diff
@@ -0,0 +1,732 @@
+
+## Table of Contents
+
+1. [Why Migrate to Versioned Workflows](#why-migrate-to-versioned-workflows)
+2. [Prerequisites](#prerequisites)
+3. [Understanding the Differences](#understanding-the-differences)
+4. [Migration Strategy](#migration-strategy)
+5. [Step-by-Step Migration](#step-by-step-migration)
+6. [Configuration Reference](#configuration-reference)
+7. [Testing and Validation](#testing-and-validation)
+8. [Common Migration Patterns](#common-migration-patterns)
+9. [Troubleshooting](#troubleshooting)
+
+## Why Migrate to Versioned Workflows
+
+If you're currently running unversioned Temporal workflows, you may be experiencing challenges with deployments:
+
+- **Deployment Risk**: Code changes can break running workflows if they're not backward compatible
+- **Rollback Complexity**: Rolling back deployments can disrupt in-flight workflows
+- **Workflow Determinism Issues**: Changes to workflow logic can cause non-deterministic errors
+
+Versioned workflows with the Temporal Worker Controller solve these problems by:
+
+- ✅ **Safe Deployments**: New versions run alongside old ones, ensuring running workflows complete successfully
+- ✅ **Automated Rollouts**: Progressive rollout strategies reduce risk of new deployments
+- ✅ **Easy Rollbacks**: Can instantly route new workflows back to previous versions
+- ✅ **Workflow Continuity**: Running workflows can continue on their original version until completion
+
+## Prerequisites
+
+Before starting the migration, ensure you have:
+
+- ✅ **Unversioned Temporal workers**: Currently running workers without Worker Versioning
+- ✅ **Kubernetes cluster**: Running Kubernetes 1.19+ with CustomResourceDefinition support
+- ✅ **Basic worker configuration**: Workers connect to Temporal with namespace and task queue configuration
+- ✅ **Administrative access**: Ability to install Custom Resource Definitions and controllers
+- ✅ **Deployment pipeline**: Existing CI/CD system that can be updated to use new deployment method
+
+### Current Environment Variables
+
+Your workers are likely configured with basic environment variables like:
+
+```bash
+TEMPORAL_HOST_PORT=your-namespace.tmprl.cloud:7233
+TEMPORAL_NAMESPACE=your-temporal-namespace
+# No TEMPORAL_DEPLOYMENT_NAME or WORKER_BUILD_ID yet
+```
+
+The controller will automatically add the versioning-related environment variables during migration.
+
+## Understanding the Differences
+
+### Before: Unversioned Workers
+
+```yaml
+# Single deployment, all workers run the same code
+apiVersion: apps/v1
+kind: Deployment
+metadata:
+  name: my-worker
+spec:
+  replicas: 3
+  template:
+    spec:
+      containers:
+      - name: worker
+        image: my-worker:v1.2.3
+        env:
+        - name: TEMPORAL_HOST_PORT
+          value: "production.tmprl.cloud:7233"
+        - name: TEMPORAL_NAMESPACE
+          value: "production"
+        # No versioning environment variables
```

Comment:
this example has no mTLS, which is fine, but do we want to demonstrate that at some point (maybe future PR) to show that aspect of the `TemporalConnection` resource?

Reply:
Added envconfig env vars, I think this is all that's needed?